seed/static/seed/js/controllers/data_quality_modal_controller.js

```diff
@@ -27,10 +27,33 @@ angular.module('BE.seed.controller.data_quality_modal', [])
       $scope.name = name;
       $scope.uploaded = moment.utc(uploaded).local().format('MMMM Do YYYY, h:mm:ss A Z');
       var originalDataQualityResults = dataQualityResults || [];
-      $scope.dataQualityResults = originalDataQualityResults;
       $scope.importFileId = importFileId;
       $scope.orgId = orgId;
 
+      var check_null = false;
+      _.forEach(originalDataQualityResults, function (results) {
+        if (results.data_quality_results.length > 1) {
+          _.forEach(_.groupBy(results.data_quality_results, 'field'), function (group) {
+            if (group.length > 1) {
+              var remove = [];
+              _.forEach(group, function (rule) {
+                if (_.isMatch(rule, {value: null}) && _.isMatch(rule, {condition: 'not_null'})) check_null = true;
+              });
+              if (check_null) {
+                _.each(group, function (rule, index) {
+                  if (rule.value === null && (rule.condition !== 'not_null' && rule.condition !== 'required')) remove.push(index);
+                });
+                for (var i = remove.length - 1; i >= 0; i--) {
+                  results.data_quality_results = _.reject(results.data_quality_results, group[remove[i]]);
+                }
+              }
+              check_null = false;
+            }
+          });
+        }
+      });
+      $scope.dataQualityResults = originalDataQualityResults;
```

Comment:
Looks like this prevents multiple not_null/required Rules from being shown. Is that right??

It looks like it's not happening on the frontend, so I think I'm misunderstanding something...

Reply:
<img width="978" alt="Screen Shot 2020-05-07 at 12 14 18 PM" src="https://user-images.githubusercontent.com/49968203/81352502-f7290e80-9083-11ea-8a00-605423f19182.png">

Reply:
In this table, there are 1 null address line 1, 4 null energy start scores, 3 null site eui, 1 null gross floor area:
![Screen Shot 2020-05-07 at 5 07 13 PM](https://user-images.githubusercontent.com/49968203/81353072-605d5180-9085-11ea-9fae-fab045b4bedf.png)

My rules are set up as:
1. check 'required' and 'not_null' address line 1;
2. check 'not null' energy star score;
3. check 'range' on enery star score;
4. check 'range' on gross floor area;
5. check 'not null' site eui;
6. check 'range' on site eui:
<img width="1213" alt="Screen Shot 2020-05-07 at 5 09 18 PM" src="https://user-images.githubusercontent.com/49968203/81353268-d1046e00-9085-11ea-9d72-96f330e541e2.png">

After running data quality check, the report should show:
1. Address line 1 is required and none for property with custom id 1 of 1;
2. Null Energy star score for property with custom id 1 of 1 (address-less property), 11 ninth, 93031 wellington, and 2660 welstone;
3. Null site eui for 11 ninth, 93031 wellington, and 2660 welstone;
4. and out of range values (site eui of 2700 welstone and energy star of 2655 welstone );
<img width="900" alt="Screen Shot 2020-05-07 at 5 12 58 PM" src="https://user-images.githubusercontent.com/49968203/81353608-bc74a580-9086-11ea-815e-b531518b4d91.png">

What is/should not being show:
1. Null energy star scores being reported 3 times for each of all properties that include null energy star scores (3 energy star score rules: 2 for ranges, 1 for not_null);
2. Null site eui being reported 3 times for all properties including null site eui values (3 site eui rules, 2 for range, 1 for not_null).